display/more.rs

```diff
@@ -0,0 +1,1963 @@
+                        let _ = self.display().inspect_err(|e| self.handle_error(*e));
+                    }else{
+                        self.prompt = Some(Prompt::More);
+                    }
+                }
+                if command == Command::UnknownCommand{ continue; } else{ remainder.clear(); }
+                self.commands_buffer = remainder;
+                let _ = self.execute(command).inspect_err(|e| self.handle_error(*e));
+                let _ = self.display().inspect_err(|e| self.handle_error(*e));
+            }
+        }
+    }
+}
+
+// If [`String`] contains existed [`PathBuf`] than returns [`PathBuf`]
+fn to_path(file_string: String) -> Result<PathBuf, MoreError>{
+    let file_string = 
+        Box::leak::<'static>(file_string.into_boxed_str());
+        let file_string = &*file_string;
+
+    let file_path = Path::new(file_string);
+    let file_path = 
+        file_path.canonicalize().map_err(|_| MoreError::FileReadError)?;
+    let _ = File::open(file_path.clone()).map_err(|_| MoreError::FileReadError)?;
+    Ok(file_path)
+}
+
+/// Get formated file name and extension from [`PathBuf`]
+fn name_and_ext(path: PathBuf) -> String {
+    let file_name = path.file_name().unwrap_or(OsStr::new("<error>"));
+    let file_name = file_name.to_str().unwrap_or("<error>");
+    format!("{}", file_name)
+}
+
+/// Format file header that can be displayed if input files count more than 1
+fn format_file_header(file_path: PathBuf, line_len: Option<usize>) -> Vec<String>{
+    let name_and_ext = name_and_ext(file_path);
+    
+    let (mut name_and_ext, border) = if let Some(line_len) = line_len{
+        let header_width = if name_and_ext.len() < 14{ 
+            14
+        } else if name_and_ext.len() > line_len - 4{
+            line_len
+        }else{
+            name_and_ext.len() + 4
+        }; 
+
+        (name_and_ext.chars().collect::<Vec<char>>()
+            .chunks(line_len)
+            .map(|ss| String::from_iter(ss))
+            .collect::<Vec<String>>(),
+        ":".repeat(header_width))
+    }else{
+        (vec![name_and_ext.clone()], ":".repeat(name_and_ext.len() + 4))
+    };
+
+    name_and_ext.insert(0, border.clone());
+    name_and_ext.push(border);
+    name_and_ext
+}
+
+/// Get char from [`&str`] by its position
+fn get_char(string: &str, position: usize) -> Option<char>{
+    string.chars().nth(position)
+}
+
+/// Parses [`String`] into [`Command`] and returns result with reminder
+fn parse(commands_str: String) -> Result<(Command, String, Command), MoreError>{
+    let mut command = Command::UnknownCommand;
+    let mut count: Option<usize> = None;
+    let mut next_possible_command = Command::UnknownCommand;
+    
+    let mut i = 0;
+    let mut chars = commands_str.chars();
+    let commands_str_len= commands_str.len();
+
+    while command == Command::UnknownCommand && i < commands_str_len{
+        let Some(ch) = chars.next() else { break; };
```

Comment:
Simplify with a `for ch in chars { .. }`
Or with `while let Some(ch) = chars.next() { .. }` if you don't want to move `chars`